plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Escrow.hs

```diff
@@ -0,0 +1,285 @@
+    (ownPubKey, vl) <- endpoint @"pay-escrow"
+    pay escrow ownPubKey vl
+
+-- | Pay some money into the escrow contract.
+pay 
+    :: HasWriteTx s
+    => EscrowParams
+    -- ^ The escrow contract
+    -> PubKey
+    -- ^ Public key of the contributor (used for refunds)
+    -> Value
+    -- ^ How much money to pay in
+    -> Contract s TxId
+pay escrow ownPubKey vl = do
+    let ds = DataScript (PlutusTx.toData ownPubKey)
+        tx = payToScript vl (escrowAddress escrow) ds
+                & validityRange .~ Ledger.interval 1 (escrowDeadline escrow)
+    writeTxSuccess tx
+
+data RedeemFailReason = DeadlinePassed | NotEnoughFundsAtAddress
+  deriving (Haskell.Eq, Show)
+
+data RedeemResult =
+    RedeemSuccess TxId
+    | RedeemFail RedeemFailReason
+    deriving (Haskell.Eq, Show)
+
+-- | 'redeem' with an endpoint.
+redeemEp 
+    :: 
+    ( HasUtxoAt s 
+    , HasAwaitSlot s
+    , HasWriteTx s
+    , HasEndpoint "redeem-escrow" () s
+    )
+    => EscrowParams
+    -> Contract s RedeemResult
+redeemEp escrow = endpoint @"redeem-escrow" >> redeem escrow
+
+-- | Redeem all outputs at the contract address using a transaction that
+--   has all the outputs defined in the contract's list of targets.
+redeem 
+    :: 
+    ( HasUtxoAt s 
+    , HasAwaitSlot s
+    , HasWriteTx s
+    )
+    => EscrowParams
+    -> Contract s RedeemResult
+redeem escrow = do
+    currentSlot <- awaitSlot 0
+    unspentOutputs <- utxoAt (escrowAddress escrow)
+    let addr = escrowAddress escrow
+        valRange = 
+            let itvl = Interval.from 1
+                -- an upper bound that does not include the escrow 
+                -- deadline
+                upperBound = Interval.UpperBound (Interval.Finite $ escrowDeadline escrow) False
+            in itvl { Interval.ivTo = upperBound }
+        tx = Typed.collectFromScriptFilter (\_ _ -> True) unspentOutputs (scriptInstance escrow) Redeem
+            & validityRange .~ valRange
+            & outputs .~ fmap mkTxOutput (escrowTargets escrow)
+    if currentSlot >= escrowDeadline escrow
+    then pure $ RedeemFail DeadlinePassed
+    else if (values unspentOutputs ^. at addr . folded) `lt` targetValue escrow
+         then pure $ RedeemFail NotEnoughFundsAtAddress
+         else RedeemSuccess <$> writeTxSuccess tx
+
+data RefundResult = RefundOK TxId | RefundFailed 
+    deriving (Haskell.Eq, Show)
+
+-- | 'refund' with an endpoint.
+refundEp 
+    :: 
+    ( HasUtxoAt s 
+    , HasWriteTx s
+    , HasEndpoint "refund-escrow" () s
+    )
+    => EscrowParams
+    -> TxId
+    -> Contract s RefundResult
+refundEp escrow txid = endpoint @"refund-escrow" >> refund escrow txid
+
+-- | Claim a refund of the contribution.
+refund 
+    :: ( HasUtxoAt s
+       , HasWriteTx s )
+    => EscrowParams 
+    -> TxId
+    -> Contract s RefundResult
+refund escrow txId = do
+    unspentOutputs <- utxoAt (escrowAddress escrow)
+    let flt TxOutRefOf{txOutRefId} _ = txId Haskell.== txOutRefId
```

Comment:
Looks like you added a `PlutusTx.Eq` instance so you probably don't need the qualifier here.